api/types/authentication.go

```diff
@@ -552,13 +567,20 @@ func (c *AuthPreferenceV2) CheckAndSetDefaults() error {
 		}
 	}
 
+	// Make sure the IdP section is populated.
 	if c.Spec.IDP == nil {
+		c.Spec.IDP = &IdPOptions{}
```

Comment:
It's unfortunate that we're not consistent with capitalization of IdP but idk what we could really do about it. I see that you're pretty consistent though with using IdP in your PRs and also we seem to have more `samlIdP` than `samlIDP`.

```
$ rg "(saml|SAML)IDP" -g '!test' -g '!*.proto' -g '!*.pb.go' -g '!*test.go' | wc -l
      44
$ rg "(saml|SAML)IdP" -g '!test' -g '!*.proto' -g '!*.pb.go' -g '!*test.go' | wc -l
     278
```

Reply:
Part of this is my fault, as I was initially naming things `IDP` when the linter complained. I'll see if I can do anything about this.